dids/src/main/kotlin/web5/sdk/dids/methods/dht/Dht.kt

```diff
@@ -0,0 +1,277 @@
+        contentType(ContentType.Application.OctetStream)
+        setBody(body)
+      }
+    }
+
+    if (!response.status.isSuccess()) {
+      val err = runBlocking { response.bodyAsText() }
+      throw RuntimeException("Error putting message to DHT: $err")
+    }
+  }
+
+  /**
+   * Gets a message from the DHT according to the Pkarr relay specification
+   * https://github.com/Nuhvi/pkarr/blob/main/design/relays.md#get
+   *
+   * @param id The z-base-32 encoded identifier of the message to get (e.g. a did:dht suffix value) [String].
+   * @return A BEP44 message [Bep44Message].
+   * @throws IllegalArgumentException if the identifier is not a z-base-32 encoded Ed25519 public key.
+   * @throws Exception if the message is not successfully retrieved from the DHT.
+   */
+  public fun pkarrGet(id: String): Bep44Message {
+    val publicKey = ZBase32.decode(id)
+    require(publicKey.size == 32) {
+      "Identifier must be a z-base-32 encoded Ed25519 public key"
+    }
+
+    val response = runBlocking { client.get("${gateway}/${id}") }
+    if (!response.status.isSuccess()) {
+      throw RuntimeException("Error getting message from DHT")
+    }
+
+    val responseBytes = runBlocking { response.readBytes() }
+    require(responseBytes.size >= 72) {
+      "Malformed response from DHT"
+    }
+    val sig = responseBytes.sliceArray(0..63)
+    val seq = ByteBuffer.wrap(responseBytes.sliceArray(64..71)).order(ByteOrder.BIG_ENDIAN).long
+    val v = responseBytes.sliceArray(72 until responseBytes.size)
+
+    return Bep44Message(v, publicKey, sig, seq)
+  }
+
+  public companion object {
+
+    /**
+     *  Creates a BEP44 Put request according to the BEP44 specification.
+     *  https://www.bittorrent.org/beps/bep_0044.html
+     *
+     *  and the Pkarr specification
+     *  https://github.com/Nuhvi/pkarr/blob/main/design/relays.md
+     *
+     *  Enforces a maximum message size of 1000 bytes and sequence value set to
+     *  the current time in milliseconds.
+     *
+     * @param manager The key manager to use to sign the message [KeyManager].
+     * @param keyAlias The alias of an Ed25519 key to sign the message with [JWK].
+     * @param message The message to publish (a DNS packet) [Message].
+     * @return A BEP44 signed message [Bep44Message].
+     * @throws IllegalArgumentException if the private key is not an Ed25519 key.
+     * @throws IllegalArgumentException if the message is empty.
+     */
+    public fun createBep44PutRequest(manager: KeyManager, keyAlias: String, message: Message): Bep44Message {
+      // get the public key to verify it is an Ed25519 key
+      val pubKey = manager.getPublicKey(keyAlias)
+      require(
+        pubKey.keyType == Ed25519.keyType &&
+          pubKey.algorithm == Ed25519.algorithm
+      ) {
+        "Must supply an Ed25519 key"
+      }
+
+      // set the sequence number to the current time in milliseconds
+      val seq = System.currentTimeMillis() / 1000
```

Comment:
is it supposed to be the current time in milliseconds or seconds? comment says millis but code is calculating seconds if i understand correctly

Reply:
yes

Reply:
😅

Reply:
ah didn't see the `Outdated` tag